### ISSUE TITLE
MD020: false positive on headings ending in "#" (C#, F#), and --fix silently drops the character

### ISSUE THREAD

**Issue by juherr** [joshrotenberg/mdbook-lint#452]
`### C#` is flagged by MD020 (and MD003), and `--fix` rewrites it to `### C`, silently changing the rendered heading from "C#" to "C". Tested with the v0.15.0 `aarch64-apple-darwin` release binary.

## 1. The heading is valid and is not a closed ATX heading

Per [CommonMark §4.2](https://spec.commonmark.org/0.31.2/#atx-headings), *"the optional closing sequence of #s must be preceded by spaces or tabs"*. In `### C#` the trailing `#` is not preceded by a space, so it is content, not a delimiter. The reference parser agrees:

```
"### C#" -> <h3>C#</h3>
```

markdownlint has the same detection false positive (DavidAnson/markdownlint#404, closed as "escape the hash"), but its prescribed workaround does not help here — `### C\#` is still flagged by mdbook-lint.

## 2. The documented "correct" examples are not headings

`docs/src/rules/standard/md020.md` lists `##Another heading##` and `###Correctly closed heading###` as ✅ correct, and `## Another heading ##` as ❌ incorrect.

That is inverted: an ATX heading requires a space *after* the opening hashes, so those "correct" examples render as paragraphs:

```
"###Correctly closed heading###" -> <p>###Correctly closed heading###</p>
```

markdownlint's MD020 is `no-missing-space-closed-atx` — it fires on `##Heading##` (the space is *missing*), which is the opposite of what this implementation flags.

## 3. `--fix` loses content

```bash
printf '# T\n\n## S\n\n### Java\n\n### C#\n\ntext\n' > t.md
mdbook-lint lint --standard-only --fix --no-backup t.md
# ### C#  ->  ### C
```

`--dry-run` emits `Warning: Could not find end position for fix at 7:7` before applying it.

This only happens with MD003 and MD020 both enabled — `--disable MD003` or `--disable MD020` leaves the line untouched — which suggests two overlapping fix ranges are being merged incorrectly.

## What the fix should produce

| form | renders as | |
| --- | --- | --- |
| `### C#` | `<h3>C#</h3>` | input |
| `### C\#` | `<h3>C#</h3>` | the only content-preserving rewrite |
| `### C #` | `<h3>C</h3>` | what markdownlint `--fix` produces |
| `### C` | `<h3>C</h3>` | what mdbook-lint `--fix` produces |

Escaping is the only rewrite that leaves the rendered document unchanged.

Ideally the rule would not fire at all — treat a trailing `#` run as a closing sequence only when it is preceded by whitespace — but if it does fire, `\#` is the correct auto-fix. The overlapping-fix-range bug looks worth guarding independently: silently dropping a character is worse than the false positive itself.
